plugins/woocommerce/tests/legacy/unit-tests/rest-api/Tests/Version2/system-status.php

```diff
@@ -21,14 +21,26 @@ class WC_Tests_REST_System_Status_V2 extends WC_REST_Unit_Test_Case {
 	 */
 	public function setUp(): void {
 		parent::setUp();
-		$this->endpoint = new WC_REST_System_Status_Controller();
-		$this->user     = $this->factory->user->create(
+		$this->user = $this->factory->user->create(
 			array(
 				'role' => 'administrator',
 			)
 		);
 	}
 
+	/**
+	 * Fetches the system status data and caches it.
+	 */
+	private function get_system_status_data() {
+		static $system_status_data = null;
```

Comment:
It looks like the V3 API test initializes `self:$user` in `wpSetUpBeforeClass`, so, we could look at doing the same thing to avoid this caching caveat? I might suggest removing this function and inlining the code in `wpSetUpBeforeClass`. We can then use `self::$system_status_report_for_user` in the tests.

What do you think a good approach might be?